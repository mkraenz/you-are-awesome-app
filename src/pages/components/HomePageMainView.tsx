import React from "react";
import { StyleSheet, View } from "react-native";
import { IPostContent } from "../../redux/IPost";
import Balloon from "./Balloon";

interface Props {
    post: IPostContent;
}

const HomePageMainView = (props: Props) => (
    <View style={localStyles.textContainer}>
        <Balloon post={props.post} />
    </View>
);

const localStyles = StyleSheet.create({
    textContainer: {
        justifyContent: "center",
        padding: 20,
        flex: 1,
    },
});

export default HomePageMainView;
