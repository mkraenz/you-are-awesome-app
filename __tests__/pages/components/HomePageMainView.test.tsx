import React from "react";
import "react-native";
// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";
import HomePageMainView from "../../../src/pages/components/HomePageMainView";

it("renders correctly", () => {
    const tree = renderer
        .create(
            <HomePageMainView
                post={{
                    author: "my-author",
                    text: "my-text",
                    country: "my-country",
                }}
            />
        )
        .toJSON();
    expect(tree).toMatchSnapshot();
});
