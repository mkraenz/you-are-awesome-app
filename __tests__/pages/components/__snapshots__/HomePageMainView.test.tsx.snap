// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "justifyContent": "center",
      "padding": 20,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#1084ff",
        "borderRadius": 20,
        "marginTop": 5,
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "white",
          "fontSize": 32,
          "paddingTop": 5,
        }
      }
    >
      my-text
    </Text>
    <Text
      style={
        Object {
          "alignContent": "flex-end",
          "color": "white",
          "paddingTop": 5,
        }
      }
    >
      my-author
       from 
      my-country
    </Text>
  </View>
</View>
`;
